Profiles

Pwma77 opened the pull request:
Let's suppose get and put work, heh.
Get obtains the data when current user is equals to actual profile "owner/user". But, I left a draft in case it needs to be compared to a property. One more thing is that I don't remember if I put just to show one or more data, so this "get" requires a bit more attention xD.
Put works perfectly, I hope so... Still, both need like a bit more "security" but so far that's it buddy. Hope we can merge them cuz are already working at this point, otherwise lemme know some changes and I'll make them maybe on Sunday or idk

Pwma77 committed:
Profiles first commit that I'm sure is not fine

Pwma77 committed:
Better visuals - fixes & ideas needed

Pwma77 committed:
Upload conflict fixes - maybe 70% visualized

BrandomRobor committed:
Test

BrandomRobor committed:
No tests

BrandomRobor committed:
TEST

BrandomRobor committed:
NO TEST

BrandomRobor committed:
Created empty file for code review

BrandomRobor committed:
Added empty JSON object

Pwma77 committed:
Some fixes I hope work - couldn't test due server

BrandomRobor committed:
Add debug config file

Pwma77 committed:
put profile works - get not yet

Pwma77 committed:
get & put working - get future changes

Pwma77 commented:
Se bugeo por el linter, pero mañana veo qué arreglo

BrandomRobor commented on src/models/profiles.py:
```diff
@@ -0,0 +1,31 @@
+
+class ProfileDoc(m.Document):
+    user = m.ReferenceField(User, reverse_delete_rule=m.CASCADE)
+    email = m.EmailField(unique=True)
```
```suggestion
```
Este campo ya no debería de existir

Pwma77 replied:
roger

BrandomRobor commented on src/models/profiles.py:
```diff
@@ -0,0 +1,31 @@
+    meta = {"collection": "profiles"}
+
+
+class ProfileData(Schema):
+    phone = String()
+    contact_email = String()
+    fb_link = String()
+    insta_link = String()
+    twit_link = String()
+
+
+class ProfileUpdate(Schema):
+    phone = String()
+    contact_email = String()
+    fb_link = String()
+    insta_link = String()
+    twit_link = String()
```
Estos esquemas son exactamente iguales, hay alguna razón para mantener ambos?

Pwma77 replied:
No realmente, los usé para probar un par de cosas y olvidé borrar el extra lol

BrandomRobor commented on .github/mapi.sarif:
```diff
@@ -0,0 +1 @@
+{}
```
Uh, un remanente de mis pruebas, necesito borrar eso

Pwma77 replied:
Bien bien

BrandomRobor commented on src/controllers/profiles.py:
```diff
@@ -0,0 +1,53 @@
+        except OperationError:
+            return ControllerStatus.ERROR
+    
+    return ControllerStatus.SUCCESS
```
Falta una nueva línea aquí, es probable que por eso falló el linter

Pwma77 replied:
Creo que sí la puse, pero hice push antes de ponerla lol

BrandomRobor commented on src/controllers/profiles.py:
```diff
@@ -0,0 +1,53 @@
+
+
+def read_prof(user_id) -> ControllerStatus:
+    user_id = auth.current_user
```
Nope, así no funciona esto
```suggestion
```

Pwma77 replied:
Va, lo borro XD

Pwma77 replied:
Recuerdo que lo puse en la ruta so creo que no lo necesito por acá (ahora que lo veo)

BrandomRobor commented on src/controllers/profiles.py:
```diff
@@ -0,0 +1,53 @@
+def read_prof(user_id) -> ControllerStatus:
+    user_id = auth.current_user
+    try:
+        # prop_owner = prop.PropertyDoc.objects.get(id=data["id"])
```
Es necesario mantener código viejo?

Pwma77 replied:
Nope, creo que lo dejé porque funcionó y no quise moverle xD
Maybe diré esto en un par de comments, qué bueno que me los recuerdas. Thanks

BrandomRobor commented on src/controllers/profiles.py:
```diff
@@ -0,0 +1,53 @@
+        return ControllerStatus.ERROR
+
+    # Here instead of current, goes owner and uncomment prop owner above
+    if str(user_id["id"]) == str(prof_id["user"].id):
```
Convertir un string a un string es un poco redundante
```suggestion
    if user_id["id"] == str(prof_id["user"].id):
```

Pwma77 replied:
Pensaba que estaba como definiendolo, así como me habías dicho lol

BrandomRobor replied:
No, definir los tipos se hace en los parámetros

BrandomRobor commented on src/controllers/profiles.py:
```diff
@@ -0,0 +1,53 @@
+    user_id = auth.current_user
+    try:
+        # prop_owner = prop.PropertyDoc.objects.get(id=data["id"])
+        prof_id = prof.ProfileDoc.objects.get()
```
Aquí falta filtrar por ID, de momento este código te devuleve el primer documento de la lista

Pwma77 replied:
Filtrar por ID me mandaba error con mongo porque no coincidían los datos, alguna recomendación?

Pwma77 replied:
De momento dejé esa parte así por si se necesitaba probar, still necesita ese fix

BrandomRobor commented on src/controllers/profiles.py:
```diff
@@ -0,0 +1,53 @@
+def update_prof(data, user_id) -> ControllerStatus:
+    prof_info = prof.ProfileDoc
+
+    if str(user_id == auth.current_user):
```
Nope, el decoder "auth" NO se debe de usar en el controlador

Pwma77 replied:
Oh no. Entonces sería más como pasarlo en la ruta, me lo traigo como parámetro y lo uso como el "user_id" no?

Pwma77 replied:
Igual sigo haciendo los cambios como me los has ido comentando

Pwma77 replied:
> Oh no. Entonces sería más como pasarlo en la ruta, me lo traigo como parámetro y lo uso como el "user_id" no?

Wait no, ya vi

Pwma77 replied:
Entonces, estoy igualando el mismo id al mismo id so es una condición useless?

BrandomRobor commented on src/controllers/profiles.py:
```diff
@@ -0,0 +1,53 @@
+
+    if str(user_id == auth.current_user):
+        try:
+            if data["phone"] == "string":
```
Uh, esta condición es realmente necesaria?

Pwma77 replied:
De momento, sí. Quería usar todos los campos, pero no encontré una forma. Está curioso porque el dato por defecto es literalmente "string", so para verificar que fuese a crear un nuevo campo no encontré otra manera. Además pensaba que si del front ponen por defecto que no tenga nada y lo guardan, está de igual forma actualizando el dato, por lo que "creo" no debería haber problema. No sé si se te pueda ocurrir alguna alternativa (directamente en el modelo poner dato inicial estaría bien, pero no encontré cómo)

BrandomRobor commented on src/controllers/profiles.py:
```diff
@@ -0,0 +1,53 @@
+    prof_info = prof.ProfileDoc
+
+    if str(user_id == auth.current_user):
+        try:
+            if data["phone"] == "string":
+                try:
+                    prof_info(
+                        **data,
+                        user=user_id
+                        ).save()
+                except OperationError:
+                    return ControllerStatus.ERROR
+            else:
+                try:
+                    prof_info.objects.update(**data)
+                except OperationError:
+                    return ControllerStatus.ERROR
+        except OperationError:
+            return ControllerStatus.ERROR
```
Un try dento de otro try? Porqué?

BrandomRobor replied:
Y acabo de ver que atrapan la misma excepción en todos, completamente innecesario

Pwma77 replied:
Ya sabes, haciendo una mezcladera hasta que funcione; como no daba errores ya no lo toqué xD
Voy a cambiarlo

Pwma77 replied:
Ah ya me acordé, te iba a preguntar por posibles escenarios para ponerlos (por motivos de seguridad), pero los terminé dejando con el mismo error. Creo que debo corregir el hecho de dejar mugrero que no uso lol

BrandomRobor commented on src/routes/profiles.py:
```diff
@@ -0,0 +1,41 @@
+    summary="Get current user's profile - contact"
+    )
+@auth_required(auth)
+def get_profile():
+    current = c.read_prof(auth.current_user)
+    return current
```
En el controlador revisas errores y retornas códigos de error, pero si no los revisas en la ruta todos los errores regresarán 200 y eso es malo

Pwma77 replied:
Hmm, creo que aquí necesito ayuda, oops

BrandomRobor commented on src/routes/profiles.py:
```diff
@@ -0,0 +1,41 @@
+
+
+@router.get("/profile")
+@output(m.ProfileData(many=True), 200)
```
No, en esta ruta sólo queremos que regrese un solo resultado
```suggestion
@output(m.ProfileData, 200)
```

Pwma77 replied:
Ah sí, again para pruebas. My bad

BrandomRobor commented on src/server.py:
```diff
@@ -17,3 +18,4 @@
 
 app.register_blueprint(auth)
 app.register_blueprint(prop)
+app.register_blueprint(prof)
```
Falta línea al final del archivo

Pwma77 replied:
Uhh se me hace que la puse después de hacer el push, orita pusheo again

Pwma77 committed:
minor code changes - just to verify - nothing new

Pwma77 reviewed (commented):
En general, el nuevo commit es para verificar si lo que me comentaste acerca de remover o cambiar, va por buen camino. No añadí nada nuevo, sólo removí; no testée, still es para que me des visto bueno/malo plz y de ahí continuar

Pwma77 commented:
> En general, el nuevo commit es para verificar si lo que me comentaste acerca de remover o cambiar, va por buen camino. No añadí nada nuevo, sólo removí; no testée, still es para que me des visto bueno/malo plz y de ahí continuar

Ah, los comentarios con los que respondí lo demás de ver si como me puedes explicar ciertas cositas, se mantienen plz

BrandomRobor reviewed (changes requested):
Mejor, pero aún no. Lo veremos en la noche

Pwma77 committed:
get profiles is done

Pwma77 committed:
before last verify - update prof

Pwma77 committed:
lemme in

Pwma77 committed:
now works - get and update profile

Pwma77 committed:
clean code

BrandomRobor committed:
Reformatted files and removed unused stuff

BrandomRobor committed:
Merged 'main'

BrandomRobor committed:
Fixed email validation bug
Now every account by default will have their registration email as their contact email

BrandomRobor reviewed (approved):
Looks güd to me

BrandomRobor committed:
Merge branch 'main' into profiles
